src/inmanta/server/bootloader.py

```diff
@@ -236,3 +243,33 @@ def load_slices(
         ctx: ApplicationContext = self._collect_slices(exts, only_register_environment_settings)
         self.feature_manager = ctx.get_feature_manager()
         return ctx
+
+    async def wait_for_db(self, db_wait_time: int):
+        """Wait for the database to be up by attempting to connect at intervals.
+
+        :param db_wait_time: Maximum time to wait for the database to be up, in seconds.
+        """
+
+        start_time = asyncio.get_event_loop().time()
+
+        # Retrieve database connection settings from the configuration
+        db_settings = {
+            "host": config.db_host.get(),
+            "port": config.db_port.get(),
+            "user": config.db_username.get(),
+            "password": config.db_password.get(),
+            "database": config.db_name.get(),
+        }
+        while True:
+            try:
+                # Attempt to create a database connection
+                conn = await asyncpg.connect(**db_settings, timeout=5)  # raises TimeoutError after 5 seconds
+                LOGGER.info("Successfully connected to the database.")
+                await conn.close()  # close the connection
+                return
+            except Exception as e:
+                LOGGER.info("Waiting for database to be up.")
```

Comment:
We may want to attach the exception to the log (it might be interesting to know why we can't connect)

There is some risk of flooding the log, but I don't see a better option